### PR TITLE
Add Package Dependencies to Reports, Part I

### PR DESCRIPTION
**Pull request by Robert-Krajcik**
This is a cleaner implementation of #723, which was left in DRAFT mode
Closes both #651 and #721

### CODE REVIEW

**Comment by Robert-Krajcik** [comment]
"suggests" will be added in Part II

File: R/mod_reportPreview.R [hunk 1]
```diff
@@ -172,6 +174,33 @@ reportPreviewServer <- function(id, selected_pkg, maint_metrics, com_metrics,
...
+                  fluidRow(
+                    column(width = 12,
+                           h5("Package Dependencies",
+                              style = "text-align: center; padding-bottom: 50px;"),
+                             metricGridUI(session$ns('dep_metricGrid'))
+                    )
+                  ),
+                  br(), br(),
+                  fluidRow(
+                    column(width = 8,
```
Comment (Jeff-Thompson12) [inline R/mod_reportPreview.R:190]: I don't understand why the table is set up to only be 2/3's of the width. I don't think it looks particularly good especially since other tables are full width. We also are not making the table 2/3's width in the downloaded reports.

I guess this also raises the question on why the table is 2/3's width on the dependency tab itself. I believe it was to incorporate the dependency plot, but that plot was scrapped.

**Comment by Jeff-Thompson12** [comment]
@Robert-Krajcik remind me, what is the suggested way to setup TinyTeX to get the PDF downloads to work?

File: R/app_server.R [hunk 2]
```diff
@@ -223,6 +223,38 @@ app_server <- function(input, output, session) {
...
+      },
+      error = function(e) {
+        msg <- paste("Detailed dependency information is not available for package", selected_pkg$name())
+        rlang::warn(msg)
+        rlang::warn(paste("info:", e))
+        dep_metrics(dplyr::tibble(package = character(0), type = character(0), name = character(0)))
+      }
+    )
+  })
+
```
Comment (Jeff-Thompson12) [inline R/app_server.R:257]: This isn't going to work for the database view tab. Instead of grabbing the dependency information for each package, this pastes the dependency information for the selected package on the sidebar and pastes it into each report.

In fact if you go straight to the database view tab and never select a package, the app will crash if you try to download a report.

File: R/app_server.R [hunk 3]
```diff
@@ -223,6 +223,38 @@ app_server <- function(input, output, session) {
     get_comm_data(selected_pkg$name())
   })
   
+  loaded2_db <- eventReactive(selected_pkg$name(), {
+    req(selected_pkg$name())
+    req(selected_pkg$name() != "-")
+    
+    dbSelect("SELECT name, version, score FROM package")
+  })
```
Comment (Jeff-Thompson12) [inline R/app_server.R:231]: I think this value should be triggered when changes to the database occur, not on package selection. Also, it would make sense to use `session$userData` instead of passing it as an argument.

File: R/mod_reportPreview.R [hunk 4]
```diff
@@ -456,7 +507,7 @@ reportPreviewServer <- function(id, selected_pkg, maint_metrics, com_metrics,
       
       tagList(
         h5(code('{riskmetric}'), 'Assessment Date:'), selected_pkg$date_added(),
-        if('Risk Score' %in% report_includes()) tagList(h5('Risk Score:'), selected_pkg$score()) else "",
+        if('Risk Score' %in% report_includes()) tagList(hr(), br(), h5('Risk Score:'), selected_pkg$score()) else "",
```
Comment (Jeff-Thompson12) [inline R/mod_reportPreview.R:510]: Why the change?

File: inst/report_downloads/reportDocx.Rmd [hunk 5]
```diff
@@ -69,19 +72,19 @@ outputComments <- function(pkg_name, comments, none_txt = "No comments"){
...
   strong('Description:'),  br(), getElement(params$pkg, 'description'),  br(), br(),
   strong('Author:'),       br(), getElement(params$pkg, 'author'),       br(), br(),
   strong('Maintainer:'),   br(), getElement(params$pkg, 'maintainer'),   br(), br(),
   strong('License:'),      br(), getElement(params$pkg, 'license'),      br(), br(),
-  strong('Published:'),    br(), getElement(params$pkg, 'published'),    br(), br(),
-  strong(code('{riskmetric}'), 'Assessment Date:'),    br(), getElement(params$pkg, 'date_added'),    br(), br(),
-  if('Risk Score' %in% params$report_includes) tagList(strong('Risk Score:'),   br(), getElement(params$pkg, 'score'),        br(), br()) else "",
-  strong('Package Decision:'), br(), ifelse(is.na(params$pkg[['decision']]), 'Pending', params$pkg[['decision']])
+  strong('Published:'),          getElement(params$pkg, 'published'),    br(), br(),
+  strong(code('{riskmetric}'), 'Assessment Date:'),          getElement(params$pkg, 'date_added'),    br(),
```
Comment (Jeff-Thompson12) [inline inst/report_downloads/reportDocx.Rmd:85]:
```suggestion
  strong(code('{riskmetric}'), 'Assessment Date:'),          getElement(params$pkg, 'date_added'),    br(), br(),
```
You took out an additional break I don't think you meant to.

File: inst/report_downloads/reportDocx.Rmd [hunk 6]
```diff
@@ -143,11 +148,11 @@ if('Maintenance Metrics' %in% params$report_includes){
       )
     ) %>%
     dplyr::select(`Metric Name`, `Metric Description`, `Metric Value`, `Metric Score`) %>%
-    knitr::kable(format = 'pandoc')
+    kableExtra::kbl(format = 'pipe', booktabs = T) 
 }
 ```
 
-`r if ('Maintenance Metrics' %in% params$report_includes) {"* Metrics whose score is NA will not impact the package {riskmetric} score"}`
+`r if ('Maintenance Metrics' %in% params$report_includes) {"Metrics whose score is NA will not impact the package {riskmetric} score"}`
```
Comment (Jeff-Thompson12) [inline inst/report_downloads/reportDocx.Rmd:155]: If we are going to remove `*` can we at least do something to make it clear this is a footer to the table.

File: inst/report_downloads/reportDocx.Rmd [hunk 7]
```diff
@@ -190,12 +197,12 @@ if (!cm_ind) {
...
       ) %>%
       dplyr::select(`Metric Name`, `Metric Description`, `Metric Value`, `Metric Score`) %>%
-      knitr::kable(format = 'pandoc')
+      kableExtra::kbl(format = 'pipe', booktabs = T)
   }
 }
 ```
 
-`r if ('Community Usage Metrics' %in% params$report_includes) {"* Metrics whose score is NA will not impact the package {riskmetric} score"}`
+`r if ('Community Usage Metrics' %in% params$report_includes) {"Metrics whose score is NA will not impact the package {riskmetric} score"}`
```
Comment (Jeff-Thompson12) [inline inst/report_downloads/reportDocx.Rmd:205]: Same as other footnote comment.

**Review by Jeff-Thompson12 — CHANGES_REQUESTED** [review]
@Robert-Krajcik Few comments below.

**Comment by Robert-Krajcik** [comment]
> @Robert-Krajcik remind me, what is the suggested way to setup TinyTeX to get the PDF downloads to work?

As mentioned in #699 
One way we can "standardize" the installation of tinytex is to have everyone just download
https://yihui.org/tinytex/TinyTeX.zip
(this is the moderately large zip file 225+ Mb that has everything we need)
and then unzip to your AppData\Roaming folder in windows.

(It would have been ideal for the bundle install to work
`tinytex::install_tinytex(bundle = 'TinyTeX')`

File: inst/report_downloads/reportDocx.Rmd (hunk 7 above)
Comment (Robert-Krajcik) [reply inst/report_downloads/reportDocx.Rmd:205]: The asterisk is currently missing from the column headers, so I will add it back in both the metric score header and the footnote.

Comment (Robert-Krajcik) [reply inst/report_downloads/reportDocx.Rmd:205]: This looks more like it
image

File: inst/report_downloads/reportDocx.Rmd (hunk 6 above)
Comment (Robert-Krajcik) [reply inst/report_downloads/reportDocx.Rmd]: asterisk is back in the footnote

File: R/mod_reportPreview.R (hunk 1 above)
Comment (Robert-Krajcik) [reply R/mod_reportPreview.R]: OK set it to width = 12

File: R/app_server.R (hunk 3 above)
Comment (Robert-Krajcik) [reply R/app_server.R]: OK using session$userData now.

File: R/mod_reportPreview.R (hunk 4 above)
Comment (Robert-Krajcik) [reply R/mod_reportPreview.R:511]: I thought it needed a horizontal rule.  reverted.

File: R/app_server.R (hunk 2 above)
Comment (Robert-Krajcik) [reply R/app_server.R]: OK I got this one moved

**Review by Jeff-Thompson12 — CHANGES_REQUESTED** [review]
In addition to the conversations below I have a few additional comments:

Why is "Metric Score" on two lines for the Maintenance Metrics table but one line on the Community Usage Metrics?
image

The HTML report hides issues with the dependency table headers that show up on the PDF DOCX outputs.
image

Also, I'm not sure if this was introduced in your PR or if this was present earlier but I noticed some funny spacing on the PDF and DOCX reports. You can see above that there is a lot of space between the header and the table. This is not the case for the Maintenance or Community Metric sections. You can also see the same behavior with Function Explorer.

File: R/mod_reportPreview.R [hunk 8]
```diff
@@ -402,6 +429,31 @@ reportPreviewServer <- function(id, selected_pkg, maint_metrics, com_metrics,
...
+    # Package Dependencies metrics cards.
+    metricGridServer("dep_metricGrid", metrics = dep_cards)
+
+    dep_table <- eventReactive(dep_metrics(), {
+      req(dep_metrics())
+
+      repo_pkgs <- as.data.frame(utils::available.packages()[,1:2])
+      
+      purrr::map_df(dep_metrics()$name, ~get_versnScore(.x, session$userData$loaded2_db(), repo_pkgs)) %>%
+      right_join(dep_metrics(), by = "name") %>%
```
[the remark this answers is not in the document]
Comment (Robert-Krajcik) [reply R/mod_reportPreview.R:458]: updated like mod_downloadHandler

File: tests/testthat/_snaps/databaseView/003.json [hunk 9]
```diff
@@ -14,7 +14,7 @@
         "1.1.2"
       ],
       "score": [
-        0.3,
+        0.25,
```
Comment (Jeff-Thompson12) [inline tests/testthat/_snaps/databaseView/003.json:17]: Do you know why this value changed for your PR? It hasn't changed on `dev`. Can you investigate why this is the case?

Comment (Robert-Krajcik) [reply tests/testthat/_snaps/databaseView/003.json:17]: I have no idea why this changed for me, but I can copy the 003.json from dev

**Review by AARON-CLARK — APPROVED** [review]
Looks good! With the understanding that you'll target #738 next. Thanks @Robert-Krajcik
